app/api/speakers.py

```diff
@@ -111,42 +92,68 @@ def before_post(self, args, kwargs, data):
         :return:
         """
         require_relationship(['event', 'user'], data)
-        kwargs['user_id'] = current_identity.id
 
-    def before_create_object(self, data, view_kwargs):
+        if not has_access('is_coorganizer', event_id=data['event']):
+            if has_access('auth_required'):
```

Comment:
I don't think ` if has_access('auth_required'): ` is required here as `is_coorganizer` already has jwt_required decorator and will check whether user is logged in or not